### PR TITLE
docs(python): Fix various typos, grammar and formatting in docstrings and user guide

### PR DESCRIPTION
**Pull request by henryharbeck**
Changes include
- Typos, grammar, punctuation, formatting
- Erroneous "underlying data, the names given here will overwrite them." line was repeated in docstrings
- Fix some "smallest" / "largest" references in `top_k` / `bottom_k`
- References to a `LazyFrame` being a `DataFrame` and vice-versa

I did notice a few more references to a `LazyFrame` being a `DataFrame` and vice-versa (for example in `LazyFrame.join`), but will leave for another PR to keep the size manageable.

### CODE REVIEW

File: docs/user-guide/concepts/data-structures.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Data structures
 
-The core base data structures provided by Polars are `Series` and `DataFrames`.
+The core base data structures provided by Polars are `Series` and `DataFrame`.
```
Comment (henryharbeck) [inline docs/user-guide/concepts/data-structures.md:3]: `LazyFrame` could also be added. Thoughts?

File: py-polars/polars/lazyframe/frame.py [hunk 2]
```diff
@@ -140,7 +140,7 @@ class LazyFrame:
         Two-dimensional data in various forms; dict input must contain Sequences,
         Generators, or a `range`. Sequence may contain Series or other Sequences.
     schema : Sequence of str, (str,DataType) pairs, or a {str:DataType,} dict
-        The DataFrame schema may be declared in several ways:
+        The LazyFrame schema may be declared in several ways:
```
Comment (henryharbeck) [inline py-polars/polars/lazyframe/frame.py:143]: Techincally, you are declaring a `DataFrame` schema when initialising a `LazyFrame` through this class, however I think this update reduces potential confusion in the reader.

File: docs/user-guide/concepts/data-structures.md (hunk 1 above)
Comment (alexander-beedie) [reply docs/user-guide/concepts/data-structures.md:3]: Well, technically LazyFrame isn't really a data structure in its own right - it's a query plan _against_ a data structure (specifically, against a DataFrame) ;)

**Comment by alexander-beedie** [comment]
This is a great collection of fixes - well spotted!✌️

**Review by alexander-beedie — APPROVED** [review]

File: docs/user-guide/concepts/data-structures.md (hunk 1 above)
Comment (henryharbeck) [reply docs/user-guide/concepts/data-structures.md:3]: Yes, true, good points. Will leave as is.

**Review by stinodego — APPROVED** [review]
Great, thanks!

**Pull request merged**